algopy_testing/src/algopy_testing/context.py

```diff
@@ -74,33 +85,106 @@ class ContractContextData:
     app_id: algopy.UInt64
 
 
+@dataclass
+class InnerTxnLoader:
+    """
+    Stores inner transaction references.
+    """
+
+    def __init__(self, inner_txn_group: Sequence[InnerTransactionResultType]):
+        self.inner_txn_group = inner_txn_group
+
+    def _get_inner_transaction(self, txn_type: type[T]) -> T:
+        txn = next(
```

Comment:
This behaviour could be misleading, would be safer to just get the last transaction and error if it isn't the required type?

Reply:
+1